### PR TITLE
implement update learning object

### PR DESCRIPTION
**Pull request by mfranz1**
# Description
This PR implements changes necessary to update a learning-object from the builder

https://github.com/Cyber4All/clark-client/assets/55514493/1c158055-7bb5-40d6-93da-5b02038a8aeb

### CODE REVIEW

File: src/app/core/user-module/user.service.ts [hunk 1]
```diff
@@ -29,9 +29,9 @@ export class UserService {
       })
       .pipe(catchError(this.handleError))
       .toPromise()
-      .then((val: any) => {
-        const arr = val;
-        return arr.map((user) => new User(user));
+      .then((res: {users: User[], total: number}) => {
+        console.log(res);
```
Comment (mfranz1) [inline src/app/core/user-module/user.service.ts:33]:
```suggestion
```

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 2]
```diff
@@ -948,8 +948,8 @@ export class BuilderStore {
       // if value is undefined here, this means we've called this function without a delay and there aren't any cached values
       // in this case we'll use the current data instead
       value = value || data;
-
-      if (!this.learningObject.id) {
+      console.log('learning-object', value);
```
Comment (mfranz1) [inline src/app/onion/learning-object-builder/builder-store.service.ts:951]:
```suggestion
```

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 3]
```diff
@@ -975,6 +975,7 @@ export class BuilderStore {
     this.learningObjectService
       .create(object)
       .then(learningObject => {
+        console.log('learning-object res', learningObject);
```
Comment (mfranz1) [inline src/app/onion/learning-object-builder/builder-store.service.ts:978]:
```suggestion
```

File: src/entity/learning-object/learning-object.ts [hunk 4]
```diff
@@ -566,11 +567,13 @@ export class LearningObject {
    * @memberof LearningObject
    */
   addContributor(contributor: User): number {
+    console.log('yo wtf', contributor);
```
Comment (mfranz1) [inline src/entity/learning-object/learning-object.ts:570]:
```suggestion
```

File: src/entity/learning-object/learning-object.ts [hunk 5]
```diff
@@ -566,11 +567,13 @@ export class LearningObject {
...
     if (contributor) {
       const addingUser =
         contributor instanceof User ? contributor : new User(contributor);
+      console.log('yo wtf', addingUser);
```
Comment (mfranz1) [inline src/entity/learning-object/learning-object.ts:574]:
```suggestion
```

File: src/app/core/user-module/user.service.ts [hunk 6]
```diff
@@ -29,9 +29,8 @@ export class UserService {
       })
       .pipe(catchError(this.handleError))
       .toPromise()
-      .then((val: any) => {
-        const arr = val;
-        return arr.map((user) => new User(user));
+      .then((res: {users: User[], total: number}) => {
```
Comment (curlyLasagna) [inline src/app/core/user-module/user.service.ts:32]: 🎨

**Review by curlyLasagna — APPROVED** [review]

**Review by shanioffe — APPROVED** [review]
Nice

File: src/app/core/user-module/user.service.ts (hunk 6 above)
Comment (johngonz0) [reply src/app/core/user-module/user.service.ts:32]: does this fix work the same as changing it to `arr.users.map`?

**Review by johngonz0 — APPROVED** [review]

**Review by pzalep1 — COMMENTED** [review]
Looks good. Just the question with _id

File: src/app/admin/components/add-evaluator/components/object-dropdown/object-dropdown.component.ts [hunk 7]
```diff
@@ -79,12 +79,12 @@ export class ObjectDropdownComponent implements OnInit, OnDestroy {
       await this.learningObjectService
         .getLearningObjects({text: query})
         .then( (results: {learningObjects: LearningObject[], total: number}) => {
-          const assignedIds = this.assignedObjects.map(obj => obj.id);
+          const assignedIds = this.assignedObjects.map(obj => obj._id);
           const filteredObjects: LearningObject[] = [];
 
           // Filters out already selected objects
           results.learningObjects.forEach( (obj: LearningObject) => {
-            if (!assignedIds.includes(obj.id) && (Array.isArray(obj.assigned) && !obj.assigned.includes(this.user.id))) {
```
[the remark this answers is not in the document]
Comment (mfranz1) [reply src/app/admin/components/add-evaluator/components/object-dropdown/object-dropdown.component.ts:87]: Yes. For some reason _id was no longer being mapped to id so just updating to _id fixed it

**Review by pzalep1 — APPROVED** [review]

**Review by stef359 — APPROVED** [review]

**Pull request merged**